packages/pwa-kit-runtime/src/ssr/server/build-remote-server.js

```diff
@@ -444,6 +445,34 @@ export const RemoteServerFactory = {
         // This method is used by the dev server, but is not needed here.
     },
 
+    /**
+     * Set up environment variables based on x-forward-* headers
+     * @private
+     */
+    _setXForwardedEnvVars(app, options) {
+        app.use((req, res, next) => {
+            const xForwardedHost = req.headers?.['x-forwarded-host']
+            if (xForwardedHost) {
+                process.env.X_FORWARDED_HOST = `${options.protocol}://${xForwardedHost}`
+            }
+
+            // since X-FORWARDED-* is attached to header on a request
+            // and process.env is a global object
+            // we only want to use the env variable when it is existing in a request
+            // once it is done, we should remove it to avoid leaking this value to other places
+            const afterResponse = () => {
+                if (process.env.X_FORWARDED_HOST) {
+                    delete process.env.X_FORWARDED_HOST
+                }
+            }
```

Comment:
Nice consideration 👍

Reply:
I don't think this will work as intended because Node.js is non-blocking when it comes to handling requests, there could be multiple requests being handled at the same time.

It's NOT a waterfall...

The code assumes:

```
request 1 received
response 1 sent

request 2 received
response 2 sent

...
```


but in reality, it could look like this
```
request 1 received
request 2 received
request 3 received

response 3 sent
response 1 sent
response 2 sent
...
```

Reply:
How do you suggest to pass the `x-forwarded-host` header to the SDK for `getAppOrigin` if it is not using process.env ? 🤔 
I believe this is the pattern we are using for `process.env.APP_ORIGIN`.

Reply:
let's think a new strategy for this...

Reply:
@kevinxh the issue of this was discussed before. One argument here is would this be considered a risk since we mostly know that the value of `x-forwarded-host` header is the origin hostname (e.g http://canadagoos.ca), which is rare to change from one request to another from one single application.

Reply:
Using `res.locals` will be helpful in this case? https://expressjs.com/en/api.html#res.locals

We've used it in this file already.

Reply:
An idea to use `res.locals`: in the retail-react-app's AppConfig, we have access to the `locals` and so we can wrap a React Context provider and save the x-forwarded header value there.

And in the pwa-kit-runtime, we can follow this pattern: https://github.com/SalesforceCommerceCloud/pwa-kit/blob/5c1f12f1c3a422dfba13d8579748726865b8044f/packages/pwa-kit-runtime/src/ssr/server/build-remote-server.js#L614-L617

Reply:
@vmarta, I've thought of that, but we are not able to access res object from within `getAppOrigin` unless you pass res object as an argument. 🤔

Reply:
Take a look at this commit. It's a draft of what I had in mind: https://github.com/SalesforceCommerceCloud/pwa-kit/commit/5872d3dd4d40c304b9b460b8382f47ba1a7c7f9e

We can create a custom React hook `useAppOrigin` that has access to the x-forwarded-host header (which is saved in a React context).

Reply:
Thanks @vmarta. That looks promising. Let me give it a test and get back to you.